api/src/main/java/com/okta/idx/sdk/api/util/ClientUtil.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.idx.sdk.api.util;
+
+import com.okta.commons.lang.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class ClientUtil {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClientUtil.class);
+
+    /**
+     * Construct the normalized URL given an issuer and a resource uri.
+     *
+     *
+     * @param issuer the issuer url
+     * @param resourceUri the uri of resource
+     * @return the normalized full url
+     */
+    public static String getNormalizedUri(String issuer, String resourceUri) throws MalformedURLException {
+        String normalizedUri;
+
+        if (isRootOrgIssuer(issuer)) {
+            // remove trailing / before concatenating with /oauth2
+            normalizedUri = issuer.replaceAll("$/", "") + "/oauth2" + resourceUri;
+        } else {
+            normalizedUri = issuer + resourceUri;
```

Comment:
Should we replace the trailing slash on issuer here as well?

Reply:
yeah, good point.. updated.